client/pages/details.jsx

```diff
@@ -4,14 +4,100 @@ export default class Details extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      currentProduct: null
+      currentProduct: null,
+      customizations: {
+        name: '',
+        date: '',
+        custom: '',
+        brand: ''
+      },
+      quantity: ''
     };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
   }
 
   componentDidMount() {
     this.getProduct();
   }
 
+  handleChange() {
+    const inputId = event.target.id;
+    const emptyObject = {};
+    const newState = Object.assign(this.state, emptyObject);
+    if (inputId === 'custom-select-brand') {
+      newState.customizations.brand = event.target.value;
+      this.setState(newState);
+    } else if (inputId === 'custom-request-brand') {
+      newState.customizations.custom = event.target.value;
+      this.setState(newState);
+    } else if (inputId === 'custom-request-name') {
+      newState.customizations.name = event.target.value;
+      this.setState(newState);
+    } else if (inputId === 'custom-request-date') {
+      newState.customizations.date = event.target.value;
+      this.setState(newState);
+    } else if (inputId === 'custom-request-custom') {
+      newState.customizations.custom = event.target.value;
+      this.setState(newState);
+    } else if (inputId === 'qty-select-custom') {
+      newState.quantity = event.target.value;
+      this.setState(newState);
+    } else if (inputId === 'qty-select-brand') {
+      newState.quantity = event.target.value;
+      this.setState(newState);
+    } else if (inputId === 'qty-select-name-date') {
+      newState.quantity = event.target.value;
+      this.setState(newState);
+    } else if (inputId === 'custom-request-name-date') {
+      newState.customizations.custom = event.target.value;
+      this.setState(newState);
```

Comment:
Ok, a few things about this section.
- Object.assign needs to be passed an empty object _first_ if you want to make a copy.
- Handle the special case first (quantity). `this.setState({ quantity: newQuantity })`. Then include a `return` statement to prevent the rest of the code from running.
- Include a `name` prop on each of your inputs. That way you can generalize the change handler for the other fields.
    ```js
    const { name, value } = event.target
    const customizationsCopy = Object.assign({}, this.state.customizations)
    customizationsCopy[name] = value
    this.setState({ customizations: customizationsCopy })
    ```